lib/dry/auto_inject/builder.rb

```diff
@@ -13,13 +13,16 @@ class Builder
       def initialize(container, options = {})
         @container = container
         @strategies = options.fetch(:strategies) { Strategies }
+      end
+
+      def method_missing(name, *args, &block)
+        return super unless strategies.key?(name)
+
+        Injector.new(container, strategies[name])
+      end
```

Comment:
I know this is minor stuff but a simple if/else is always easier to parse with eyes than `unless` with return so:

``` ruby
if strategies.key?(name)
  Injector.new(container, strategies[name])
else
  super
end
```